### PR TITLE
Hotfix color sorting

### PR DESCRIPTION
**Pull request by zonky2**
Hotfix colorSort

shot197

###################

shot198

### CODE REVIEW

File: src/MetaModels/Attribute/Color/Color.php [hunk 1]
```diff
@@ -110,12 +111,17 @@ public function sortIds($idList, $strDirection)
...
+    private function colorSort($colors, $strDirection)
     {
-        $counter = 0;
-        $sorted  = array();
+        $counter  = 0;
+        $sorted   = array();
+        $unsorted = array();
         foreach ($colors as $itemId => $colorValue) {
             $condensed = $colorValue[0];
+            if ($condensed == '') {
```
Comment (discordier) [inline src/MetaModels/Attribute/Color/Color.php]: Why do you keep track of the empty colors separately?
They will then not get sorted correctly.

File: src/MetaModels/Attribute/Color/Color.php [hunk 2]
```diff
@@ -110,12 +111,17 @@ public function sortIds($idList, $strDirection)
      *
      * @return array
      */
-    private function colorSort($colors)
+    private function colorSort($colors, $strDirection)
```
Comment (discordier) [inline src/MetaModels/Attribute/Color/Color.php]: Variable name should be only `$direction`, we are not using prefixes in new code.
Additionally, the parameter is not documented (as already indicated by the travis inspection).

File: src/MetaModels/Attribute/Color/Color.php [hunk 3]
```diff
@@ -126,8 +132,13 @@ private function colorSort($colors)
             $counter++;
         }
 
-        ksort($sorted);
-        return $sorted;
+        if ($strDirection === 'desc') {
```
Comment (discordier) [inline src/MetaModels/Attribute/Color/Color.php]: Should be:
```php
if ('DESC' === $direction) {
  krsort($sorted);
} else {
  ksort($sorted);
}
```
In Uppercase (See `IAttribute.php` comment).

**Comment by zonky2** [comment]
without merge_array
shot199

**Comment by discordier** [comment]
I have implemented your suggestions in the current alpha2 branch (ecb4183d33379fcea8c9f01b7be4fbacc58e5048 and b0f245ea870ea3b883b0fc4850821fc60e2fdaf8), could you please test again?
The unit test works as expected.

**Comment by zonky2** [comment]
works fine (after Hotfix https://github.com/MetaModels/core/blob/hotfix-2.0.0-alpha11/src/MetaModels/DcGeneral/Data/Driver.php#L387)

shot201

**Pull request closed without merging**